dimod/reference/samplers/exact_solver.py

```diff
@@ -159,6 +159,45 @@ def sample_poly(self, polynomial, **kwargs):
         """
         return ExactSolver().sample(polynomial, **kwargs)
 
+class ExactSolverDQM(Sampler):
+    """A simple exact solver for testing and debugging code using your local CPU.
+
+    Notes:
+        This solver becomes slow for problems with 18 or more
+        variables, worse for increasingly many cases per variable.
+
+    """
+    properties = None
+    parameters = None
+
+    def __init__(self):
+        self.properties = {}
+        self.parameters = {}
+
+    def sample_dqm(self, dqm, **kwargs):
+        """Sample from a discrete quadratic model.
+
+        Args:
+            dqm (:obj:`~dimod.DiscreteQuadraticModel`):
+                Discrete quadratic model to be sampled from.
+
+        Returns:
+            :obj:`~dimod.SampleSet`
+
+        """
+        kwargs = self.remove_unknown_kwargs(**kwargs)
+        
+        
+        n = dqm.num_variables()
+        if n == 0:
+            return SampleSet.from_samples([], 'DISCRETE', energy=[])
+
+        possible_samples = as_samples(_all_cases_dqm(dqm))
+        energies = dqm.energies(possible_samples[0])
+        
+        response = SampleSet.from_samples(possible_samples, 'DISCRETE', energies).relabel_variables(dict(enumerate(dqm.variables)))
```

Comment:
rather than relabeling, you can just pass in `SampleSet.from_samples((samples, labels), 'DISCRETE',...`